File: Python/Algorithms/1608.py
class Solution:
    def specialArray(self, nums):
        length = len(nums)
        rng = range(length)
        # Start from end. (todo, try starting from beginning)
        for i in range(1, length+1):
            # keep track of matches (>= i)
            match = 0
            for idx in rng:
                n = nums[idx]
                if n >= i:
                    match = match + 1
                    # bail when matches are > i.
                    if match > i:
                        break
                    continue
                # Bail if not enough elements left.
                if match + length - idx < i:
                    break
            # return as soon as we find the match.
            if match == i:
                return i
        return -1

File: C/Algorithms/1608.c
int specialArray(int *nums, int numsSize) {
  for (int i = 1; i < numsSize + 1; i++) {
    int match = 0;
    for (int j = 0; j < numsSize; j++) {
      int el = nums[j];
      if (el >= i) {
        match++;
        if (match > i)
          break;
        continue;
      }
      // break if num of elements left to see
      if (match + (numsSize - j) < i)
        break;
    }
    if (match == i)
      return i;
  }
  return -1;
}
